system-tests/api-changing-settings-should-dump/src/test/java/com/teamscale/tia/client/ApiChangingSettingsShouldDumpSystemTest.java

```diff
@@ -0,0 +1,33 @@
+package com.teamscale.tia.client;
+
+import com.teamscale.test.commons.SystemTestUtils;
+import com.teamscale.test.commons.TeamscaleMockServer;
+import org.junit.jupiter.api.Test;
+import systemundertest.SystemUnderTest;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Runs the system under test and then changes the partition of the agent. This should cause a dump to our
+ * {@link TeamscaleMockServer}.
+ */
+public class ApiChangingSettingsShouldDumpSystemTest {
+
+	@Test
+	public void systemTest() throws Exception {
+		System.setProperty("org.eclipse.jetty.util.log.class", "org.eclipse.jetty.util.log.StdErrLog");
+		System.setProperty("org.eclipse.jetty.LEVEL", "OFF");
+
+		TeamscaleMockServer teamscaleMockServer = new TeamscaleMockServer(
+				SystemTestUtils.TEAMSCALE_PORT).acceptingReportUploads();
+
+		new SystemUnderTest().foo();
+		SystemTestUtils.changePartition(SystemTestUtils.AGENT_PORT, "some_other_value");
+
+		assertThat(teamscaleMockServer.uploadedReports).hasSize(1);
```

Comment:
Should we check that the report was uploaded under the old partition name as expected?

Reply:
good idea